### ISSUE TITLE
Bug: file ignored in flat config is still parsed by the parser

### ISSUE THREAD

**Issue by timmywil** [eslint/eslint#17411]
### Environment

Node version: v18.16.1
npm version: v9.5.1
Local ESLint version: v8.45.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.19045

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
{
  files: [
    "test/**"
  ],
  ignores: [
    "test/data/json_obj.js"
  ]
}
```
</details>

```js
{ "data": {"lang": "en", "length": 25} }
```

I'm currently in the process of migrating jQuery's fairly complicated eslint config to use flat config instead. Other than some issues I'm having with the import plugin, this is the only issue left. I want to avoid changing the test or JS file, which means I need eslint not to parse it at all.

### What did you expect to happen?

I don't think the JS should still be parsed. The file can be correctly ignored in the config, but the parser is still parsing it and throwing a syntax error. There are valid use cases for ignoring invalid JS, bypassing the parser altogether, and currently no **flat** configuration, including examples above, will fix that. Besides, it's also a performance concern. Seems like a waste of resources to parse a file that isn't going to be linted.

### What actually happened?

```sh
error  Parsing error: Unexpected token :
```

Changing the file to be valid JS does make it clear the file is still ignored by the linter, so `ignores` is set properly, but I don't want it to even try to parse the file. This worked fine before flat config.

### Link to Minimal Reproducible Example

https://github.com/timmywil/eslint-test

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

I was able to work around it for now, by setting the parser to return an empty AST.

```js
{
  files: [ "test/data/json_obj.js" ],
  languageOptions: {
    parser: {
      meta: {
        name: "Ignore Without Parsing"
      },

      // Ignore invalid JS
      parse: function() {
        return {
          type: "Program",
          loc: {},
          comments: [],
          range: [
            0,
            0
          ],
          body: [],
          tokens: []
        };
      }
    }
  }
}
```

**Comment by aladdin-add** [comment]
this is working as expected - as explained in https://github.com/eslint/eslint/issues/17400#issuecomment-1646543502.
